src/unaudited/ECDSAStakeRegistry.sol

```diff
@@ -415,32 +446,38 @@ contract ECDSAStakeRegistry is
     /**
      * @notice Common logic to verify a batch of ECDSA signatures against a hash, using either last stake weight or at a specific block.
      * @param digest The hash of the data the signers endorsed.
-     * @param operators A collection of addresses that endorsed the data hash.
+     * @param signers A collection of signing key addresses that endorsed the data hash.
      * @param signatures A collection of signatures matching the signers.
      * @param referenceBlock The block number for evaluating stake weight; use max uint32 for latest weight.
      */
     function _checkSignatures(
         bytes32 digest,
-        address[] memory operators,
+        address[] memory signers,
         bytes[] memory signatures,
         uint32 referenceBlock
     ) internal view {
-        uint256 signersLength = operators.length;
-        address currentOperator;
-        address lastOperator;
-        address signer;
+        uint256 signersLength = signers.length;
+        address currentSigner;
+        address lastSigner;
+        address operator;
         uint256 signedWeight;
-
+        
         _validateSignaturesLength(signersLength, signatures.length);
         for (uint256 i; i < signersLength; i++) {
-            currentOperator = operators[i];
-            signer = _getOperatorSigningKey(currentOperator, referenceBlock);
-
-            _validateSortedSigners(lastOperator, currentOperator);
-            _validateSignature(signer, digest, signatures[i]);
-
-            lastOperator = currentOperator;
-            uint256 operatorWeight = _getOperatorWeight(currentOperator, referenceBlock);
+            currentSigner = signers[i];
+            operator = _getOperatorForSigningKey(currentSigner, referenceBlock);
+            if (operator == address(0)) {
+                revert SignerNotRegistered();
+            }
+            if (!_operatorRegistered[operator]) {
```

Comment:
This wasn't checked in the original, maybe not here as well?
I mean, if we are checking signatures on an old block when the operator was active, but then i deregistered in the meantime? `_operatorRegistered` isn't a history. We can remove this I think (unless you have a good reason - it wasn't checked in the original)